tests/test_modeling_tf_common.py

```diff
@@ -574,7 +591,11 @@ def check_hidden_states_output(config, inputs_dict, model_class):
                 self.model_tester, "expected_num_hidden_layers", self.model_tester.num_hidden_layers + 1
             )
 
-            hidden_states = outputs[-1]
+            if self.is_encoder_decoder:
+                hidden_states = outputs.decoder_hidden_states
+            else:
+                hidden_states = outputs.hidden_states
```

Comment:
Sorry, I meant something a bit more like this: 

```python
            if model.config.is_encoder_decoder:
                encoder_hidden_states = outputs.encoder_hidden_states
                decoder_hidden_states = outputs.decoder_hidden_states
                
                self.assertEqual(config.output_attentions, False)
                self.assertEqual(len(encoder_hidden_states), expected_num_layers)
                self.assertListEqual(
                    list(encoder_hidden_states[0].shape[-2:]),
                    [self.model_tester.seq_length, self.model_tester.hidden_size],
                )
                self.assertEqual(len(decoder_hidden_states), expected_num_layers)
                self.assertListEqual(
                    list(decoder_hidden_states[0].shape[-2:]),
                    [self.model_tester.seq_length, self.model_tester.hidden_size],
                )
            else:
                hidden_states = outputs.hidden_states
                self.assertEqual(config.output_attentions, False)
                self.assertEqual(len(hidden_states), expected_num_layers)
                self.assertListEqual(
                    list(hidden_states[0].shape[-2:]),
                    [self.model_tester.seq_length, self.model_tester.hidden_size],
                )
```

1) Always good to check for both encoder & decoder for `encoder_decoder` and 2) I prefer to stop using the `self.is_encoder_decoder` flag (the config better defines whether a model is an encoder-decoder - not really the test case IMO)

Reply:
Ok, just pushed the fix!!!